src/stacked_borrows/mod.rs

```diff
@@ -914,8 +909,17 @@ trait EvalContextPrivExt<'mir: 'ecx, 'tcx: 'mir, 'ecx>: crate::MiriInterpCxExt<'
                         alloc_range(base_offset, size),
                     );
                     stacked_borrows.for_each(range, dcx, |stack, dcx, exposed_tags| {
-                        stack.grant(orig_tag, item, &global, dcx, exposed_tags)
-                    })
+                        stack.grant(orig_tag, item, access, &global, dcx, exposed_tags)
+                    })?;
+                    drop(global);
+                    if let Some(access) = access {
+                        assert!(access == AccessKind::Read);
```

Comment:
If I recall correctly, `assert_eq!` produces better panic messages, so I always use it when possible

Reply:
It prints what the values actually were. But here there are only two options.

But yeah probably better to get in the habit of using assert_eq.